### PR TITLE
[Fix] Add comment to the NewMessageNotificationBuilder

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

- comment to the NewMessageNotificationBuilder
- test that checks notifications for ephemeral messages

### CODE REVIEW

File: Source/Notifications/Push notifications/Notification Types/Content/ZMLocalNotification+Events.swift [hunk 1]
```diff
@@ -313,6 +313,7 @@ private class NewMessageNotificationBuilder: EventNotificationBuilder {
 
     override func shouldCreateNotification() -> Bool {
         guard let message = message else {
+            //We should create a notification for the ephemeral message, but we might not be able to create a generic message from the event payload.
```
Comment (johnxnguyen) [inline Source/Notifications/Push notifications/Notification Types/Content/ZMLocalNotification+Events.swift:316]: Do you know why we may not be able to create the generic message?

Comment (KaterinaWire) [reply Source/Notifications/Push notifications/Notification Types/Content/ZMLocalNotification+Events.swift]: I changed it

File: Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests_Event.swift [hunk 2]
```diff
@@ -508,7 +508,22 @@ final class ZMLocalNotificationTests_Event: ZMLocalNotificationTests {
...
+
+        // when
+        note = ZMLocalNotification(event: event, conversation: self.oneOnOneConversation, managedObjectContext: self.uiMOC)
 
+        // then
+        XCTAssertNotNil(note)
+        XCTAssertEqual(note!.title, "Super User")
+        XCTAssertEqual(note!.body, "New message")
+        
+    }
```
Comment (johnxnguyen) [inline Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests_Event.swift:525]: We already have a test for generating a notification for an unknown message, you can delete this.

File: Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests.swift [hunk 3]
```diff
@@ -174,6 +174,19 @@ class ZMLocalNotificationTests: MessagingTest {
...
+            "id": UUID.create().transportString(),
+            "conversation": conversationID.transportString(),
+            "from": senderID.transportString(),
+            "time": Date().transportString(),
+            "data": ["text": ""],
+            "type": "conversation.otr-message-add"
+        ]
+        
+        return ZMUpdateEvent(fromEventStreamPayload: payload as ZMTransportData, uuid: nonce)!
+    }
```
Comment (johnxnguyen) [inline Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests.swift:188]: Technically this is not an ephemeral message. Seeing as we don't need the test (see other comment), I think we can delete this too.

Comment (KaterinaWire) [reply Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests.swift]: Removed

File: Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests_Event.swift (hunk 2 above)
Comment (KaterinaWire) [reply Tests/Source/Notifications/PushNotifications/ZMLocalNotificationTests_Event.swift]: Removed

**Review by johnxnguyen — APPROVED** [review]

**Comment by johnxnguyen** [comment]
@KaterinaWire we're going to change the logic here to not display notifications for unknown message and instead improve the logging. We can close this PR.

**Pull request closed without merging**